inc/customizer/layouts.php

```diff
@@ -0,0 +1,498 @@
+		add_action( 'init',              array( $this, 'post_type_support' ), 11 );
+
+		add_action( 'admin_enqueue_scripts', array( $this, 'admin_enqueue_scripts' ) );
+		add_action( 'load-post.php',         array( $this, 'load_post' ) );
+		add_action( 'load-post-new.php',     array( $this, 'load_post' ) );
+		add_action( 'save_post',             array( $this, 'save_post' ) );
+
+		add_action( 'customize_register', array( $this, 'customize_register' ) );
+
+		add_filter( 'body_class', array( $this, 'body_class' ) );
+
+	}
+
+	/**
+	 * Apply theme support args.
+	 *
+	 * @action after_setup_theme
+	 * @since  1.0.0
+	 */
+	public function theme_support() {
+
+		$args    = (array) get_theme_support( 'primer-layouts' );
+		$layouts = array_keys( $this->layouts );
+
+		$this->default  = ( ! empty( $args[0] ) && $this->layout_exists( $args[0] ) ) ? $args[0] : ( $this->layout_exists( $this->default ) ? $this->default : array_shift( $layouts ) );
+		$this->meta_box = ! empty( $args[1] ) ? (bool) $args[1] : $this->meta_box;
+
+	}
+
+	/**
+	 * Alter some registered layouts when in RTL mode.
+	 *
+	 * @action init
+	 * @since  1.0.0
+	 */
+	public function rtl_layouts() {
+
+		if ( ! is_rtl() ) {
+
+			return;
+
+		}
+
+		/**
+		 * Filter changes needed for registered layouts when in RTL mode.
+		 *
+		 * @since 1.0.0
+		 *
+		 * @var array
+		 */
+		$rtl_layouts = (array) apply_filters( 'primer_layouts_rtl',
+			array(
+				'two-column-default'    => esc_html__( 'Two Columns: Sidebar | Content', 'primer' ),
+				'two-column-reversed'   => esc_html__( 'Two Columns: Content | Sidebar', 'primer' ),
+				'three-column-default'  => esc_html__( 'Three Columns: Sidebar | Sidebar | Content', 'primer' ),
+				'three-column-reversed' => esc_html__( 'Three Columns: Content | Sidebar | Sidebar', 'primer' ),
+			)
+		);
+
+		$this->layouts = array_merge( $this->layouts, $rtl_layouts );
+
+	}
+
+	/**
+	 * Add post type support.
+	 *
+	 * @action init
+	 * @since  1.0.0
+	 */
+	public function post_type_support() {
+
+		/**
+		 * Filter the post types that allow layouts.
+		 *
+		 * @since 1.0.0
+		 *
+		 * @var array
+		 */
+		$post_types = (array) apply_filters( 'primer_layouts_post_types', get_post_types( array( 'public' => true ) ) );
+
+		foreach ( $post_types as $post_type ) {
+
+			add_post_type_support( $post_type, 'primer-layouts' );
+
+		}
+
+	}
+
+	/**
+	 * Enqueue scripts and styles for post meta box.
+	 *
+	 * @action admin_enqueue_scripts
+	 * @since  1.0.0
+	 *
+	 * @param string $hook
+	 */
+	public function admin_enqueue_scripts( $hook ) {
+
+		if ( ! in_array( $hook, array( 'post.php', 'post-new.php' ) ) ) {
+
+			return;
+
+		}
+
+		$suffix = SCRIPT_DEBUG ? '' : '.min';
+
+		wp_enqueue_script( 'primer-layouts-meta-box', get_template_directory_uri() . "/assets/js/meta-box{$suffix}.js", array( 'jquery' ), PRIMER_VERSION );
+
+		wp_enqueue_style( 'primer-layouts-meta-box', get_template_directory_uri() . '/assets/css/meta-box.css', array(), PRIMER_VERSION );
```

Comment:
@schrapel I get your point about source maps but since core is made that way I think we should stick with it. It's easy enough to change SCRIPT_DEBUG to false and use source map ourselves but other devs might be used to SCRIPT_DEBUG.